paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/FlowControllerTest.kt

```diff
@@ -12,27 +12,40 @@ import com.stripe.android.networktesting.RequestMatchers.not
 import com.stripe.android.networktesting.RequestMatchers.path
 import com.stripe.android.networktesting.RequestMatchers.query
 import com.stripe.android.networktesting.testBodyFromFile
+import com.stripe.android.paymentsheet.utils.ManualIdlingResource
 import com.stripe.android.paymentsheet.utils.assertCompleted
 import com.stripe.android.paymentsheet.utils.assertFailed
 import com.stripe.android.paymentsheet.utils.runFlowControllerTest
-import org.junit.Ignore
+import org.junit.After
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
 @RunWith(AndroidJUnit4::class)
-@Ignore("Flakey, https://jira.corp.stripe.com/browse/RUN_MOBILESDK-2640")
 internal class FlowControllerTest {
     @get:Rule
     val composeTestRule = createAndroidComposeRule<MainActivity>()
 
+    private val manualIdlingResource = ManualIdlingResource()
+
     private val page: PaymentSheetPage = PaymentSheetPage(composeTestRule)
 
     @get:Rule
     val networkRule = NetworkRule()
 
+    @Before
+    fun setup() {
+        composeTestRule.registerIdlingResource(manualIdlingResource)
+    }
+
+    @After
+    fun reset() {
+        composeTestRule.unregisterIdlingResource(manualIdlingResource)
```

Comment:
If one test fails, it'll cause the rest to fail as well. We should reset the manualIdlingResource to false after every test.